### PR TITLE
LPS-117963 Fix portal-workflow-web classes

### PR DESCRIPTION
**Pull request by eduardoallegrini**
Originally this part was using the property `hidden="true"` but since this property seems to not be available in Clay/DXP anymore I prefered to upgrade it with a simple `d-none` class

**Result:**

ezgif com-video-to-gif

### CODE REVIEW

File: modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/js/main.js [hunk 1]
```diff
@@ -163,13 +163,11 @@ AUI.add(
...
-
-					if (hidden) {
-						element.removeAttribute('hidden');
+					if (element.classList.contains('d-none')) {
+						element.classList.remove('d-none');
 					}
 					else {
-						element.setAttribute('hidden', true);
+						element.classList.add('d-none');
 					}
```
Comment (eduardoallegrini) [inline modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/js/main.js:171]: I also tried with
```js
if (element.hasClass('d-none')) {
    element.removeClass('d-none');
}
else {
    element.addClass('d-none');
}
```

after reading these docs
- https://alloyui.com/api/classes/A.Node.html#method_removeClass
- https://alloyui.com/api/files/yui3_src_node_js_node-class.js.html#l24

but it didn't work

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

File: modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/js/main.js (hunk 1 above)
Comment (jbalsas) [reply modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/js/main.js:171]: There's better ways to do this in AUI, but here you don't have `Y.Node` but regular `HTMLElements` coming from down in a `document.getElementById(arguments[index]);`, so those methods are not available there.

I think this should be fine for now as it is.

**Comment by eduardoallegrini** [comment]
> Hey @eduardoallegrini, this seems okay to me for a quick fix, but I worry we probably have many other places with `hidden="true"` relying on this.

I agree but I think `hidden="true"` it's "legacy" code and we should upgrade it anyways

> Did we remove that from `Clay` or from our `Compat Layer`?

I'm not sure because it's the same we noticed during the [Clay's update](https://github.com/liferay-frontend/liferay-portal/pull/83#issuecomment-656160245) and we didn't modify the compat-layer (as far as I know)
> I feel we should probably re-add this to avoid further regressions in other places. Either that or confirm that we don't have `hidden="true"` anywhere else and mark this as a breaking change after identifying when and where it happened.

The [other issue](https://issues.liferay.com/browse/LPS-116592) you noticed previously it's caused by the same problem (compat-layer) but I couldn't find other `hidden=` usages in the portal

> Can you take a closer look to see if you can figure this out?

ok

> I think I recently saw some conversations with @pat270 about this... 🤔

**Comment by marcoscv-work** [comment]
Is this related to accessibility hidden in some way? it should be related?
The `aria` propertie is: `aria-hidden="true"`

**Comment by marcoscv-work** [comment]
I don't think we should need to create a pattern right now, but probably we need a clear way to identify:
- Something we want to hide but keep it readable by a screen reader (`sr-only` class)
- Something totally hidden (sr and visual)
- Something was totally hidden and now is visible:
  - does it need the focus?
  - does it need to annunce it appears? (`aria-live="polite"`?)

**Comment by eduardoallegrini** [comment]
> Is this related to accessibility hidden in some way? it should be related?
> The `aria` propertie is: `aria-hidden="true"`

`hidden="true"` was defined as "d-none"

image

**Comment by eduardoallegrini** [comment]
> I don't think we should need to create a pattern right now, but probably we need a clear way to identify:
> 
> * Something we want to hide but keep it readable by a screen reader (`sr-only` class)
> * Something totally hidden (sr and visual)
> * Something was totally hidden and now is visible:
>   
>   * does it need the focus?
>   * does it need to annunce it appears? (`aria-live="polite"`?)

Maybe we should open an issue in https://github.com/liferay/liferay-frontend-guidelines ?

**Comment by pat270** [comment]
@jbalsas @eduardoallegrini We remove this from Clay at https://github.com/liferay/clay/pull/3271#pullrequestreview-416692356. I believe it will be fixed with yesterday's Clay update. See https://github.com/liferay/clay/issues/3533.

**Comment by jbalsas** [comment]
Thanks for the reference @pat270!!

Ironic that I myself said:

> jbalsas on 22 May
> This seems dangerous... but IE10- 🤷

Let's verify if that fixes this but keep this going so we can eventually get rid of `hidden` more comfortably.

**Comment by eduardoallegrini** [comment]
About the PR, are these test failures related?

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
This is green, so...

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**